### PR TITLE
Added support for go1.23, bumped the minimum version to 1.22

### PR DESCRIPTION
**Pull request by codeboten**

### CODE REVIEW

File: .chloggen/codeboten_go-1.23.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: all
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Added support for go1.23, bumped the minimum version to 1.22
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (codeboten) [inline .chloggen/codeboten_go-1.23.yaml:13]:
```suggestion
issues: [10869]
```

**Review by songy23 — APPROVED** [review]

**Review by songy23 — COMMENTED** [review]
`check-links` failed though doesn't seem related to this change https://github.com/open-telemetry/opentelemetry-collector/actions/runs/10374150084/job/28720957270?pr=10869

**Comment by codeboten** [comment]
@songy23 opened a pr here: https://github.com/open-telemetry/opentelemetry-collector/pull/10874

**Review by TylerHelmuth — COMMENTED** [review]
There is a `1.22.4` lurking in `build-and-test-arm.yml`. Also a 

```
    "constraints": {
      "go": "1.21"
    },
```

in renovate.json

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**